lua/lualine/components/filetype.lua

```diff
@@ -29,11 +29,14 @@ function M:apply_icon()
   end
 
   local icon, icon_highlight_group
+  local iconft, icon_highlight_groupft
   local ok, devicons = pcall(require, 'nvim-web-devicons')
   if ok then
     icon, icon_highlight_group = devicons.get_icon(vim.fn.expand('%:t'))
-    if icon == nil then
-      icon, icon_highlight_group = devicons.get_icon_by_filetype(vim.bo.filetype)
+    iconft, icon_highlight_groupft = devicons.get_icon_by_filetype(vim.bo.filetype)
+    if iconft ~= nil then
+      icon = iconft
+      icon_highlight_group = icon_highlight_groupft
```

Comment:
```suggestion
    icon, icon_highlight_group = devicons.get_icon_by_filetype(vim.bo.filetype)
    if iconft == nil then
      icon, icon_highlight_group = devicons.get_icon(vim.fn.expand('%:t'))
    end
```

Reply:
This suggestion tried on https://github.com/nvim-lualine/lualine.nvim/pull/1186/commits/670f318907b165bf279d410ca7cf1c968249aa2b but, it failed tests.